src/lib/Renderer.js

```diff
@@ -975,92 +975,89 @@ var Renderer = Base.extend('Renderer', {
             } else if (isHeaderRow || isFooterRow) { // no checkbox on "totals" rows
-                cellProperties.value = '';
+                config.value = '';
             } else {
-                cellProperties.value = [images.checkbox(isRowSelected), rowNum, null];
+                config.value = [images.checkbox(isRowSelected), rowNum, null];
             }
         } else {
             // set dataRow and columnName used by valOrFunc (needed when func)
-            var column = behavior.getActiveColumn(c);
-            cellProperties.dataRow = grid.getRow(r);
-            cellProperties.columnName = column.name;
-            cellProperties.calculator = column.calculator;
+            config.dataRow = grid.getRow(r);
+            config.columnName = column.name;
+            config.calculator = column.calculator;
 
-            cellProperties.value = grid.getValue(c, r);
+            config.value = grid.getValue(c, r);
         }
 
-        cellProperties.isGridColumn = isGridColumn;
-        cellProperties.isGridRow = isGridRow;
-        cellProperties.isColumnHovered = grid.isColumnHovered(c) && isGridColumn;
-        cellProperties.isRowHovered = grid.isRowHovered(r) && isGridRow;
-        cellProperties.isCellHovered = grid.isHovered(c, r) && isGridColumn && isGridRow;
-        cellProperties.bounds = this._getBoundsOfCell(c, r);
-        cellProperties.isCellSelected = isCellSelected;
-        cellProperties.isRowSelected = isRowSelected;
-        cellProperties.isColumnSelected = isColumnSelected;
-        cellProperties.isInCurrentSelectionRectangle = grid.isInCurrentSelectionRectangle(c, r);
+        config.isGridColumn = isGridColumn;
+        config.isGridRow = isGridRow;
+        config.isColumnHovered = grid.isColumnHovered(c) && isGridColumn;
+        config.isRowHovered = grid.isRowHovered(r) && isGridRow;
+        config.isCellHovered = grid.isHovered(c, r) && isGridColumn && isGridRow;
+        config.bounds = this._getBoundsOfCell(c, r);
+        config.isCellSelected = isCellSelected;
+        config.isRowSelected = isRowSelected;
+        config.isColumnSelected = isColumnSelected;
+        config.isInCurrentSelectionRectangle = grid.isInCurrentSelectionRectangle(c, r);
 
         if (grid.mouseDownState) {
             var point = grid.mouseDownState.gridCell;
-            cellProperties.mouseDown = point.x === c && point.y === r;
+            config.mouseDown = point.x === c && point.y === r;
         }
 
-        var cell = behavior.getCellRenderer(cellProperties, c, r);
-
-        behavior.cellPropertiesPrePaintNotification(cellProperties);
+        var cell = behavior.getCellRenderer(config, c, r);
 
-        //declarative cell properties
-        if (isGridRow && isGridColumn) {
-            var overrides = behavior.getCellProperties(column.index, r);
-            _(cellProperties).extendOwn(overrides);
+        if (config.reapplyCellProperties) {
+            _(config).extendOwn(cellProperties);
```

Comment:
Two changes here:

0. We no longer restrict cell prop objs to grid cells only.
0. We now do this only on request because the cell props are already in `config`: Only when explicitly needed to override modifications made to `config` by `getCell`.